wat

```diff
@@ -1,6 +1,10 @@
 #!/bin/sh
 log_lines() {
-  local regex='^.*\(starting full system upgrade\|upgraded.*\)$'
+  local installed=''
+  if [ "x$SHOW_INSTALLED" == "x1" ]; then
```

Comment:
I wouldn't call *passing* an argument *parsing* an argument. I would do something like this:

```sh
log_lines() {
  local show_installed="$1" regex

  if [ $show_installed -eq 1 ]; then
    regex=...
  else
    regex=...
  fi

  sed ...
}

show_installed=0

case "$1" of
   -i|--installed) show_installed=1 ;;
   *) err_usage ;; # meh, or just ignore, I dunno
esace

log_lines "$show_installed | ...
```